product_unique_serial/wizard/stock_serial.py

```diff
@@ -79,31 +80,33 @@ def default_get(self, field_list):
 
     @api.onchange('serial_ids')
     def onchange_serial(self):
-        serial = []
-
-        for serial_name in self.serial_ids:
-            if serial_name.serial in serial:
-                return {
-                    'warning': {
-                        'title': _('Warning'),
-                        'message': _(
-                            'The Serial number {} already captured'.format(
-                                serial_name.serial.encode('utf-8')))
-                    }}
-            else:
-                serial.append(serial_name.serial)
+        serials = [serial.serial for serial in self.serial_ids]
```

Comment:
with a `mapped` i don't get repeated items, for that reason i use a comprehension list, to be able to get repeated items and display error message